client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/dynamic-settings.js

```diff
@@ -0,0 +1,75 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+
+/**
+ * Internal dependencies
+ */
+import {
+	getCarrierRegistrationFields
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/actions';
+import * as api from 'woocommerce/woocommerce-services/api';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import DynamicCarrierAccountSettingsForm from './dynamic-settings-form';
+
+export const DynamicCarrierAccountSettings = ( props ) => {
+	const [carrierRegistrationFields, setCarrierRegistrationFields] = useState([]);
+
+	useEffect(() => {
+		const fetchRegistrationFields = async () => {
+			const registrationFields = await api.get(props.siteId, api.url.shippingCarrierTypes());
+			setCarrierRegistrationFields(registrationFields.carriers);
+		}
+		fetchRegistrationFields();
+	}, [props.siteId]);
+
+	if (!carrierRegistrationFields || carrierRegistrationFields.length < 1) {
+		return (
+			<div>Loading...</div>
+		);
+	}
+
+	const supportedCarriers = carrierRegistrationFields.filter(carrier => carrier.type === props.carrier);
+
+	if (!supportedCarriers || supportedCarriers.length === 0) {
+		return (
+			<div>{props.carrier} not supported.</div>
+		);
+	}
+
+	const currentCarrierRegistrationField = supportedCarriers[0];
+
+    return (
+		<div>
+			<DynamicCarrierAccountSettingsForm
+				carrierType={currentCarrierRegistrationField.type}
+				carrierName={currentCarrierRegistrationField.name}
+				registrationFields={currentCarrierRegistrationField.fields}
+			/>
+		</div>
+	);
+};
+
+const mapStateToProps = ( state ) => {
+	return {
+		siteId: getSelectedSiteId( state ),
+    };
+};
+
+const mapDispatchToProps = ( dispatch, {siteId} ) => ( {
+	getCarrierRegistrationFields: () => dispatch( getCarrierRegistrationFields( siteId ) ),
```

Comment:
There should be no need to connect this, since it's not dispatching any Redux actions.

Reply:
Yes, we should not need this anymore. Thanks.